corehq/apps/enterprise/api/resources.py

```diff
@@ -363,26 +364,25 @@ class Meta(ODataEnterpriseReportResource.Meta):
 
     REPORT_SLUG = EnterpriseReport.FORM_SUBMISSIONS
 
-    def get_report_task(self, request):
-        enddate = datetime.strptime(request.GET['enddate'], '%Y-%m-%d') if 'enddate' in request.GET else None
-        startdate = datetime.strptime(request.GET['startdate'], '%Y-%m-%d') if 'startdate' in request.GET else None
+    def get_object_list(self, request):
+        # TODO: logic to handle when the start/end date are null or last 30 days
+        start_date = request.GET.get('start_date', None)
+        end_date = request.GET.get('end_date', None)
+        last_domain = request.GET.get('last_domain', None)
+        last_time = request.GET.get('last_time', None)
+        last_id = request.GET.get('last_id', None)
+
         account = BillingAccount.get_account_by_domain(request.domain)
-        return generate_enterprise_report.s(
-            self.REPORT_SLUG,
-            account.id,
-            request.couch_user.username,
-            start_date=startdate,
-            end_date=enddate,
-            include_form_id=True,
-        )
+
+        return get_enterprise_form_iterator(account, start_date, end_date, last_domain, last_time, last_id)
```

Comment:
Could the page size limit be passed in here? Might be able to retrieve that with something like
```py
limit = self.paginator.get_limit()
```